core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -266,62 +265,68 @@ export class RteHelperModalComponent {
       let rawLatex: string = value[0].raw_latex;
       let mathExpressionSvgIsBeingProcessed: boolean = (
         value[0].mathExpressionSvgIsBeingProcessed);
-      this.saveButtonIsDisabled = (
-        mathExpressionSvgIsBeingProcessed || rawLatex === '');
+      if (mathExpressionSvgIsBeingProcessed || rawLatex === '') {
+        this.updateRteErrorMessage(' ');
+      } else {
+        this.updateRteErrorMessage('');
+      }
     } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
       let start: number = value[1];
       let end: number = value[2];
       if (start === 0 && end === 0) {
-        this.saveButtonIsDisabled = false;
-      } else {
-        this.saveButtonIsDisabled = (start >= end);
-      }
-      if (this.saveButtonIsDisabled) {
-        this.updateRteErrorMessage(
-          'Please ensure that the start time of the video is earlier than ' +
-          'the end time.');
-      } else {
         this.updateRteErrorMessage('');
+      } else {
+        if (start >= end) {
+          this.updateRteErrorMessage(
+            'Please ensure that the start time of the video is earlier than ' +
+            'the end time.');
+        } else {
+          this.updateRteErrorMessage('');
+        }
       }
     } else if (this.componentId === this.COMPONENT_ID_LINK) {
       let url: string = value[0];
       let text: string = value[1];
       if (text === '') {
         text = url;
```

Comment:
With regards to [this comment](https://github.com/oppia/oppia/pull/19179#issuecomment-1859275176), here I think you can set the form `value[1]` to match the URL too, if we don't do that already elsewhere. (No need to change anything else below.)

Reply:
done in the [PR](https://github.com/oppia/oppia/pull/20071/files#diff-300d28d3f0d817f80f59012396737dd2d20ced8984eee02ec750c60c4683bf7a)